explain-joins.md

```diff
@@ -0,0 +1,267 @@
+| │   └─TableFullScan_20       | 142020.00 | 71010   | cop[tikv] | table:t1      | proc max:104ms, min:28ms, p80:104ms, p95:104ms, iters:83, tasks:3                                                                                                                                                                                        | keep order:false                                  | N/A                   | N/A     |
+| └─TableReader_24(Probe)      | 180000.00 | 90000   | root      |               | time:356.586006ms, loops:90, cop_task: {num: 3, max: 357.264333ms, min: 301.721986ms, avg: 326.268422ms, p95: 357.264333ms, max_proc_keys: 31712, p95_proc_keys: 31712, tot_proc: 536ms, rpc_num: 3, rpc_time: 978.759747ms, copr_cache_hit_ratio: 0.00} | data:TableFullScan_23                             | 178.60922050476074 MB | N/A     |
+|   └─TableFullScan_23         | 180000.00 | 90000   | cop[tikv] | table:t2      | proc max:100ms, min:88ms, p80:100ms, p95:100ms, iters:102, tasks:3                                                                                                                                                                                       | keep order:false                                  | N/A                   | N/A     |
++------------------------------+-----------+---------+-----------+---------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------+-----------------------+---------+
+6 rows in set (0.36 sec)
+
+Query OK, 0 rows affected (3.16 sec)
+```
+
+### Variations of Index Join
+
+An index join using the hint [`INL_JOIN`](/optimizer-hints.md#inl_joint1_name--tl_name-) will create a hash table of the intermediate results before joining on the outer table. TiDB also supports creating a hash table on the outer table with the hint [`INL_HASH_JOIN`](/optimizer-hints.md#inl_hash_join). If the column sets on the inner table match the columns of the outer table, the [`INL_MERGE_JOIN`](/optimizer-hints.md#inl_merge_join) index join can apply. Each of these variations of Index Join will be automatically selected by the SQL Optimizer.
+
+### Configuration
+
+Index Join performance is influenced by the following system variables:
+
+* [`tidb_index_join_batch_size`](/system-variables.md#tidb_index_join_batch_size) (default: 25000) - the batch size of index lookup operations.
+* [`tidb_index_lookup_join_concurrency`](/system-variables.md#tidb_index_lookup_join_concurrency) (default: 4) - the number of concurrent index lookup tasks.
+
+## Hash Join
```

Comment:
Should `Hash Join` be above on `Index Join` because the example in `Index Join` relies on `Hash Join`?

Reply:
I think because MySQL only had index join until recently, it is better to have index join first. It even says in the second sentence "This method of join works similar to the primary method of join used in MySQL.".

There are two sentences which may be hard to parse:

> Index Join is effective if the build side is small and the probe side is pre-indexed and large. Consider the following query where the index join performs much better than the hash join:

I could reword this and say "where the index join performs well".

> If the index on `t1.int_col` is not available, the hash join execution plan remains the same. However, the plan changes for the Index Join and performance is reduced:

This is in reference to an example, where it can be seen the first example uses INL_JOIN and the second uses HASH_JOIN. I think it is understandable in context that hash join will be explained after.